kotsadm/pkg/updatechecker/updatechecker.go

```diff
@@ -199,33 +203,14 @@ func CheckForUpdates(appID string, deploy bool) (int64, error) {
 		Silent:              false,
 	}
 
-	// get info about the deployed app version for reporting purposes
-	downstreams, err := store.GetStore().ListDownstreamsForApp(a.ID)
+	// add info for reporting purposes
+	r, err := GetReportingInfo(a.ID)
 	if err != nil {
-		return 0, errors.Wrap(err, "failed to list downstreams for app")
-	}
-
-	deployedAppSequence, err := downstream.GetCurrentParentSequence(a.ID, downstreams[0].ClusterID)
-	if err != nil {
-		return 0, errors.Wrap(err, "failed to get current downstream parent sequence")
-	}
-
-	if deployedAppSequence != -1 {
-		deployedArchiveDir, err := store.GetStore().GetAppVersionArchive(a.ID, deployedAppSequence)
-		if err != nil {
-			return 0, errors.Wrap(err, "failed to get app version archive")
-		}
-
-		deployedKotsKinds, err := kotsutil.LoadKotsKindsFromPath(deployedArchiveDir)
-		if err != nil {
-			return 0, errors.Wrap(err, "failed to load kotskinds from path")
-		}
-
-		getUpdatesOptions.DownstreamCursor = deployedKotsKinds.Installation.Spec.UpdateCursor
-		getUpdatesOptions.DownstreamChannelID = deployedKotsKinds.Installation.Spec.ChannelID
-		getUpdatesOptions.DownstreamChannelName = deployedKotsKinds.Installation.Spec.ChannelName
+		return 0, errors.Wrap(err, "failed to get reporting info")
```

Comment:
I don't think this should fail completely. Log this and pass `nil` to the ReportingInfo field so that the installation continues to function.